src/sentry/snuba/metrics/datasource.py

```diff
@@ -276,38 +302,70 @@ def get_metrics_meta(
 
 def get_stored_metrics_of_projects(
     projects: Sequence[Project],
-    use_case_id: UseCaseID,
+    use_case_ids: Sequence[UseCaseID],
     start: datetime | None = None,
     end: datetime | None = None,
 ) -> Mapping[str, Sequence[int]]:
     org_id = projects[0].organization_id
     project_ids = [project.id for project in projects]
 
-    stored_metrics = []
-    entity_keys = get_entity_keys_of_use_case_id(use_case_id=use_case_id)
-    for entity_key in entity_keys or ():
-        stored_metrics += _get_metrics_by_project_for_entity(
-            entity_key=entity_key,
-            project_ids=project_ids,
-            org_id=org_id,
-            use_case_id=use_case_id,
-            start=start,
-            end=end,
+    # We compute for each use case, the entities that we want to query.
+    entity_keys = defaultdict(set)
+    for use_case_id in use_case_ids:
+        entity_keys[use_case_id] = entity_keys[use_case_id].union(
+            get_entity_keys_of_use_case_id(use_case_id=use_case_id)
         )
 
-    grouped_stored_metrics = {}
-    for stored_metric in stored_metrics:
-        grouped_stored_metrics.setdefault(stored_metric["metric_id"], []).append(
-            stored_metric["project_id"]
-        )
+    # We compute a list of all the queries that we want to run in parallel across entities and use cases.
+    requests = []
+    use_case_id_to_index = defaultdict(list)
+    for use_case_id, entity_keys in entity_keys.items():
+        for entity_key in entity_keys:
+            requests.append(
+                _get_metrics_by_project_for_entity_query(
+                    entity_key=entity_key,
+                    project_ids=project_ids,
+                    org_id=org_id,
+                    use_case_id=use_case_id,
+                    start=start,
+                    end=end,
+                )
+            )
+            use_case_id_to_index[use_case_id].append(len(requests) - 1)
```

Comment:
nit: i would do these two things in the same nested loops

Reply:
Which two things?

Reply:
computing  entity_keys dict and then computing a list of requests

Reply:
I did put them separate to make the implementation work even if you pass two use case ids and if entity keys are different, but it's a bit overengineered. I can do like you said, will simplify! Thanks for the suggestion